app/models/entry.server.ts

```diff
@@ -123,38 +128,49 @@ export function getEntriesByInitialLetters(
   >`SELECT id, headword FROM det_entries WHERE LOWER(headword) LIKE LOWER(${initialLettersWildcard}) ORDER BY LOWER(headword) ASC LIMIT ${take} OFFSET ${skip}`
 }
 
-export function getEntriesByBasicTextSearchAndPage(
-  text: string,
-  page: string = "1",
-  caseSensitive: boolean = false
+export async function updateRecordByAttributeAndType(
+  type: attributeEnum,
+  id: number,
+  value: string
 ) {
-  const pageNumber = parseInt(page)
-  if (isNaN(pageNumber)) {
-    throw new Error(`Page Number ("${page}") must be a number`)
-  } else if (isNonPositive(pageNumber)) {
-    throw new Error(`Page Number ("${page}") must be greater than zero`)
+  if (isNaN(id)) {
+    throw new Error(`Error Parsing ID of element being edited`)
+  } else if (isNonPositive(id)) {
+    throw new Error(`Error Parsing Type and ID of element being edited`)
   }
 
-  const skip: number = (pageNumber - 1) * DEFAULT_PAGE_SIZE
-  return getEntriesByBasicTextSearch(text, skip, undefined, caseSensitive)
+  switch (type) {
+    case attributeEnum.HEADWORD:
+      await updateEntryHeadword(id, value)
+      break
+    case attributeEnum.ETYMOLOGY:
+      await updateEntryEtymology(id, value)
+      break
+    case attributeEnum.LABLES:
+      await updateEntryLables(id, value)
+      break
+    default:
+      throw new Error("Type of element being edited is not supported")
+  }
 }
 
-export function getEntriesByBasicTextSearch(
-  text: string,
-  skip: number = 0,
-  take: number = 100,
-  caseSensitive: boolean = false
-) {
-  if (text.length === 0) {
-    throw new Error(`Text ("${text}") length must be greater than zero`)
-  }
-  const searchWildcard = `%${text}%`
+export async function updateEntryHeadword(entryId: number, newValue: string) {
```

Comment:
Just leaving this here as a comment for now, you don't have to fix this in this PR, but maybe in the future:

The structure of the `prisma.entry.update` call is very similar. It's always:

```js
    await prisma.entry.update({
     where: { id: entryId },
     data: { someValueName: newValue },
   })
```

Because of this, instead of doing the `switch` statement up above and selectively calling a function, we could do something like this:

```js
const TYPE_MAP = {
  [attributeEnum.HEADWORD]: "headword",
  [attributeEnum.ETYMOLOGY]: "etymology",
  [attributeEnum.LABELS]: "general_labels",
}
```

...and then have a general function for updating:

```js
export async function updateEntry(entryId: number, attributeType: attributeEnum, newValue: string) {
    await prisma.entry.update({
     where: { id: entryId },
     data: { [TYPE_MAP[attributeType]]: newValue },
   })
}
```

...and then instead of the switch statement we just go:

```js
await updateEntry(id, type, value)
```

I'm sure there would be some other cleaning up of the code required (e.g. error handling) but this would be a bit tidier, I think. NB: I have just written the code off the top of my head, it might have bugs 😃

Reply:
yeah good idea! I'll Submit a PR for this